### PR TITLE
Elasticsearch: headers support for retrieving data

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR addresses the feature request: https://github.com/tensorflow/io/issues/1093 by allowing the users to pass headers while connecting to and fetching data from elasticsearch clusters.

This implies the users can now pass the authentication header and fetch data from a protected cluster.

### CODE REVIEW

**Comment by kvignesh1420** [comment]
cc: @yongtang  @terrytangyuan

File: tensorflow_io/core/python/experimental/elasticsearch_dataset_ops.py [hunk 1]
```diff
@@ -106,7 +116,7 @@ def get_healthy_resource(self):
                 continue
         else:
             raise ConnectionError(
-                "No healthy node available for this index, check the cluster status and index"
+                "No healthy node available for this index, please check the cluster config"
```
Comment (terrytangyuan) [inline tensorflow_io/core/python/experimental/elasticsearch_dataset_ops.py:119]: Consider printing out the healthcheck url/index

File: tests/test_elasticsearch/elasticsearch_test.sh [hunk 2]
```diff
@@ -37,22 +43,34 @@ docker run -d --rm --name=tfio-elasticsearch \
...
+echo ""
 echo "Waiting for the elasticsearch cluster to be up and running..."
+echo ""
 sleep 20
 
+echo ""
 echo "Checking the base REST-API endpoint..."
-curl localhost:9200/
+echo ""
+curl -X GET localhost:9200/ --header 'Authorization: Basic ZWxhc3RpYzpkZWZhdWx0X3Bhc3N3b3Jk'
```
Comment (terrytangyuan) [inline tests/test_elasticsearch/elasticsearch_test.sh:54]: Where did `ZWxhc3RpYzpkZWZhdWx0X3Bhc3N3b3Jk` come from?

File: tensorflow_io/core/python/experimental/elasticsearch_dataset_ops.py (hunk 1 above)
Comment (kvignesh1420) [reply tensorflow_io/core/python/experimental/elasticsearch_dataset_ops.py]: addressed in the commit.

File: tests/test_elasticsearch/elasticsearch_test.sh (hunk 2 above)
Comment (kvignesh1420) [reply tests/test_elasticsearch/elasticsearch_test.sh:56]: addressed in the commit.

Comment (kvignesh1420) [reply tests/test_elasticsearch/elasticsearch_test.sh:56]: I am setting the password for the `elastic` user using env vars. So it is the base64 encoded string of "elastic:default_password". Used only for testing purposes.

**Review by terrytangyuan — APPROVED** [review]
LGTM. Others PTAL

**Review by yongtang — APPROVED** [review]
LGTM

**Pull request merged**